main.tf

```diff
@@ -89,11 +89,16 @@ resource "google_storage_bucket_object" "main" {
 }
 
 // todo(bharathkkb): remove workaround after https://github.com/hashicorp/terraform-provider-google/issues/11383
+// Also: https://github.com/hashicorp/terraform/issues/28925 (when this functions project is created)
 data "google_project" "nums" {
-  for_each   = toset(concat(compact([for item in var.secret_environment_variables : lookup(item, "project_id", "")]), [var.project_id]))
+  for_each   = toset(compact([for item in var.secret_environment_variables : lookup(item, "project_id", "")]))
   project_id = each.value
 }
 
+data "google_project" "main" {
```

Comment:
Can we only create this if any secrets exist?
```
count = length(var.secret_environment_variables) > 0 ? 1 : 0
```

Reply:
I think not, because doing just that would break:

```
try(data.google_project.nums[secret_environment_variables.value["project_id"]].number, data.google_project.default.number)
```

Reply:
Not sure I follow can't we just do 
```
try(data.google_project.nums[secret_environment_variables.value["project_id"]].number, data.google_project.default[0].number)
```
IIUC we use this just for secrets and length of `secret_environment_variables` should be known at apply time even if values are not?

Reply:
Sorry, missed that. Updated the PR.